src/pybind/mgr/alert/module.py

```diff
@@ -0,0 +1,194 @@
+                'default': None
+            },
+            {
+                'name': 'twitter_access_token'
+                'default': None
+            },
+            {
+                'name': 'twitter_access_token_secret'
+                'default': None
+            }
+    ]
+
+    COMMANDS = [
+        {
+            "cmd": "email alert "
+                   "name=status_code,type=CephString,req=false",
+            "desc": "Send an email alert to inform user of their Ceph cluster status",
+            "perm": "r"
+        },
+        {
+            "cmd": "slack alert ",
+            "desc": "post to slack",
+            "perm": "r"
+        }
+    ]
+
+    def __init__(self, *args, **kwargs):
+        super(Module, self).__init__(*args, **kwargs)
+        # set up some members to enable the serve() method and shutdown
+        self.run = True
+        self.event = Event()
+
+    def init_module_config(self):
+        self.config['mail_server'] = \
+            self.get_config("mail_server", default=self.config_keys['mail_server'])
+        self.config['mail_port'] = \
+            int(self.get_config("mail_port", default=self.config_keys['mail_port']))
+        self.config['email_receiver'] = \
+            self.get_config("email_receiver", default=self.config_keys['email_receiver'])
+        self.config['email_sender'] = \
+            self.get_config("email_sender", default=self.config_keys['email_sender'])
+        self.config['sender_password'] = \
+            self.get_config("sender_password", default=self.config_keys['sender_password'])
+        self.config['slack_api'] = \
+            int(self.get_config("slack_api", default=self.config_keys['slack_api']))
+        self.config['twitter_consumer_key'] = \
+            int(self.get_config("twitter_consumer_key", default=self.config_keys['twitter_consumer_key']))
+        self.config['twitter_consumer_secret'] = \
+            int(self.get_config("twitter_consumer_key", default=self.config_keys['twitter_consumer_key']))
+        self.config['twitter_access_token'] = \
+            int(self.get_config("twitter_access_token", default=self.config_keys['twitter_access_token']))
+        self.config['twitter_access_token_secret'] = \
+            int(self.get_config("twitter_access_token_secret", default=self.config_keys['twitter_access_token_secret']))
+
+    def get_api():
+        auth = tweepy.OAuthHandler(self.config['twitter_consumer_key'], self.config['twitter_consumer_secret'])
+        auth.set_access_token(slef.config['twitter_access_token'], self.config['twitter_access_token_secret'])
+        return tweepy.API(auth)
+
+    def tweets(alert_message):
+        api = get_api()
+        tweet = alert_message
+        status = api.update_status(status=tweet)
+        return status
+
+    def slack(alert_message):
+        slack = Slacker(self.config['slack_api'])
+        post = alert_message
+        return post
+
+    def email(alert_message):
```

Comment:
I would make this take both ``subject`` and ``message`` arguments

Reply:
The reason why I had only `message` argument is because twitter, slack and irc doesn't  support subject. My thinking is to put all these skeletons in a single hepler function that is called to send the alert to the different platforms

Reply:
yeah, makes sense.  upon further reflection it maybe makes more sense to have an option 'detail' argument for the email case. probably simplest not to worry about that for now and add it in later!